File: src/py-opentimelineio/opentimelineio/schema/marker.py
# SPDX-License-Identifier: Apache-2.0
# Copyright Contributors to the OpenTimelineIO project

from .. core._core_utils import add_method
from .. import _otio


@add_method(_otio.Marker)
def __str__(self):
    return "Marker({}, {}, {}, {})".format(
        str(self.name),
        str(self.marked_range),
        str(self.color),
        str(self.metadata),
    )


@add_method(_otio.Marker)
def __repr__(self):
    return (
        "otio.schema.Marker("
        "name={}, "
        "marked_range={}, "
        "color={}, "
        "metadata={}"
        ")".format(
            repr(self.name),
            repr(self.marked_range),
            repr(self.color),
            repr(self.metadata),
        )
    )

File: src/opentimelineio/marker.cpp
// SPDX-License-Identifier: Apache-2.0
// Copyright Contributors to the OpenTimelineIO project

#include "opentimelineio/marker.h"
#include "opentimelineio/missingReference.h"

namespace opentimelineio { namespace OPENTIMELINEIO_VERSION_NS {

Marker::Marker(
    std::string const&   name,
    TimeRange const&     marked_range,
    std::string const&   color,
    AnyDictionary const& metadata,
    std::string const&   comment)
    : Parent(name, metadata)
    , _color(color)
    , _marked_range(marked_range)
    , _comment(comment)
{}

Marker::~Marker()
{}

bool
Marker::read_from(Reader& reader)
{
    return reader.read_if_present("color", &_color)
           && reader.read("marked_range", &_marked_range)
           && reader.read_if_present("comment", &_comment)
           && Parent::read_from(reader);
}

void
Marker::write_to(Writer& writer) const
{
    Parent::write_to(writer);
    writer.write("color", _color);
    writer.write("marked_range", _marked_range);
    writer.write("comment", _comment);
}

}} // namespace opentimelineio::OPENTIMELINEIO_VERSION_NS
